coalib/output/ConsoleInteraction.py

```diff
@@ -725,16 +725,20 @@ def try_to_apply_action(action_name,
                                          file_dict,
                                          file_diff_dict,
                                          section)
-        if not isinstance(chosen_action, DoNothingAction):
-            console_printer.print(
-                format_lines(chosen_action.SUCCESS_MESSAGE, symbol='['),
-                color=SUCCESS_COLOR)
-        applied_actions[action_name] = [copy.copy(result), copy.copy(
-            file_dict),
-                                    copy.copy(file_diff_dict),
-                                    copy.copy(section)]
-        result.set_applied_actions(applied_actions)
-        failed_actions.discard(action_name)
+        if len(file_diff_dict) == 0 and isinstance(chosen_action,
+                                                   IgnoreResultAction):
+            failed_actions.add(action_name)
+        else:
+            if not isinstance(chosen_action, DoNothingAction):
+                console_printer.print(
+                    format_lines(chosen_action.SUCCESS_MESSAGE, symbol='['),
+                    color=SUCCESS_COLOR)
+            applied_actions[action_name] = [copy.copy(result), copy.copy(
+                file_dict),
+                                        copy.copy(file_diff_dict),
+                                        copy.copy(section)]
```

Comment:
Maybe hang the opening and closing square brackets on new lines? I don't exactly know pep8 guidelines for this but do check it.